### DISCUSSION TITLE
Why do we need endpoints ?

### DISCUSSION THREAD

**Discussion by NewUniverseEntrop** [vercel/next.js#13850] [Help] [answered]
Hey, just want to know for next.js, is it necesary to build api endpoints for login etc ? 

why not directly work with database ?  why do we need to fetch(" ") data when we can just get it by doing queries ... 

https://softwareengineering.stackexchange.com/questions/277701/why-do-people-do-rest-apis-instead-of-dbals

Thanks !

**Comment by Timer** [comment, marked as answer]
It is not necessary to expose endpoints. You can directly query databases from `getStaticProps` and `getServerSideProps`.

See https://next-code-elimination.now.sh/ for an example and [this docs page](https://nextjs.org/docs/basic-features/data-fetching) for more info!
